### PR TITLE
REALMC-12450: Update onboarding docs for M1 users

### PR DESCRIPTION
**Pull request by NBSquare**
[Ticket](https://jira.mongodb.org/browse/REALMC-12450)

This one isn't blocked by the stitch support lib ticket but we can wait until the [baas PR](https://github.com/10gen/baas/pull/6261) is merged if we want to close out the ticket all at once.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -94,6 +94,7 @@ To lint the project, run:
 ```cmd
 golangci-lint run
 ```
+> Note: `golangci-lint` seems to panic on M1 machines with a `can't load fmt` error. This issue is documented [here](https://github.com/golangci/golangci-lint/issues/2374). While this is still being fixed, it seems that installing `golangci-lint` from source resolves the issue.
```
Comment (makesitgo) [inline README.md:97]: oh I meant to ask about this on the other PR but:

what does "seem to panic" mean?  is it flaky?

**Review by makesitgo — APPROVED** [review]
🆒  beans

File: README.md (hunk 1 above)
Comment (NBSquare) [reply README.md:97]: Not the best wording here... Meant to say that it panics and it seems like it's an M1 issue. I'll clarify this.

**Review by tcannon91 — APPROVED** [review]

**Comment by NBSquare** [comment]
Merging as evergreen failures are unrelated to this patch

**Pull request merged**